synapse/handlers/federation.py

```diff
@@ -1651,15 +1652,9 @@ async def _sync_partial_state_room(
 
         # Make an infinite iterator of destinations to try. Once we find a working
         # destination, we'll stick with it until it flakes.
-        destinations: Collection[str]
-        if initial_destination is not None:
-            # Move `initial_destination` to the front of the list.
-            destinations = list(other_destinations)
-            if initial_destination in destinations:
-                destinations.remove(initial_destination)
-            destinations = [initial_destination] + destinations
-        else:
-            destinations = other_destinations
+        destinations = _prioritise_destinations_for_partial_state_resync(
```

Comment:
if we're factoring out this logic... wouldn't it make sense to move the constraint check just above (`if initial_destination is None and len(other_destinations) == 0:`) out too?

Reply:
Sure.

FWIW: my motivation for pulling it out is that `_sync_partial_state_room` is a long[^1] function and this helps to keep it short. Looking back on the change though: it's pretty arguable. No strong opinions.